client/state/data-layer/wpcom/videos/poster/index.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { VIDEO_EDITOR_UPDATE_POSTER } from 'state/action-types';
+import {
+	setPosterUrl,
+	showError,
+	showUploadProgress,
+} from 'state/ui/editor/video-editor/actions';
+
+/**
+ * Updates the poster for a video.
+ *
+ * @param  {Object} store  Redux store
+ * @param  {Object} action  Action object
+ * @param {Function} next  Dispatches to next middleware in chain
+ * @returns {Object} original action
+ */
+export const updatePoster = ( { dispatch }, action, next ) => {
+	if ( ! ( 'file' in action.params ) && ! ( 'at_time' in action.params ) ) {
```

Comment:
just for fun, we can also flip this boolean expression and to some people (me included) it would be more obvious what we're checking. to others it won't be. in general, there will be times where it can dramatically simplify the logic. for more info, check out boolean reduction and _deMorgan's_ law

```js
if ( ! ( 'file' in action.params || 'at_time' in action.params ) ) {
	return next( action );
}
```

further, if we know we would want to use these anyway, we can actually go ahead and extract them in a way which yet again can lead to more insight into our goals

```js
const { at_time, file } = pick( action.params, [ 'at_time', 'file' ] );

if ( ! ( at_time && file ) ) {
	return next( action );
}
```

now these values are already bound for us